src/index.ts

```diff
@@ -34,59 +34,67 @@ export const remarkMdxFrontmatter: Attacher<[RemarkMdxFrontmatterOptions?]> =
       );
     }
 
-    for (const node of mdast.children) {
-      let data: unknown;
-      const { value } = node as YAML;
-      if (node.type === 'yaml') {
-        data = load(value);
-        // @ts-expect-error A custom node type may be registered for TOML frontmatter data.
-      } else if (node.type === 'toml') {
-        data = parse(value);
-      }
-      if (data == null) {
-        return;
-      }
```

Comment:
Because of this line, the front matter isn't working since it never reaches the line that adds the import due to return from the function.

Reply:
`remark-frontmatter` has support for multiple frontmatter nodes, so the loop is necessary. The fix would be to revert the changes you made in this file, but change this `return` statement to a `continue`.

Reply:
hah! I knew it hehe :D 

Maybe do a `filter` then.

Reply:
Help me here, even thou that is true, does the exiting code would work thou?

Reply:
I changed it from `.forEach()` to `for (const … of …)` a while ago. That’s why it broke. `continue` inside a loop is equivalent to `return` inside ``forEach()`.

> Help me here, even thou that is true, does the exiting code would work thou? the declaration always use the same `name` so I would assume it would overwrite or blow up, no?

When using the `name` property, yes. Without it, it could still work. For example:

```md
---
foo: bar
---

Hello

---
fooz: baz
---

World
```